### ISSUE TITLE
directory in the prompt has not been updated when the $PWD renamed

### ISSUE THREAD

**Issue by haolian9** [romkatv/powerlevel10k#2304]
to reproduce:

* `p10k configure` as below (with `transient prompt` on especially)
* `$ mkdir foo`
* `$ cd foo` -> `/tmp/foo >`
* `$ mv ../foo ../bar`
* `$ cd ..` -> `/tmp >`
* `$ cd bar` -> `/tmp/foo >`

as you can see (if you tried on your side too, lol), at the last step, we are in `bar/`, while the prompt still shows `foo`.

i have no other plugin/scripts around zsh, just powerlevel10k, so i think this could be a problem of it.

my env:
* archlinux
* zsh 5.9 from archlinux's extra repo
* p10k: 1.16.1 from archlinux's community repo
* st: 0.9 from aur

my `p10k configure`:
* look like a diamond? no
* look like a lock? no
* look like a lock? no
* look like `><`? no
* prompt style? lean
* prompt color? 8
* show time? no
* prompt height? one line
* prompt spacing? compact
* prompt flow? concise
* transient prompt? yes
* instant prompt mode? verbose
* overwrite? no

**Comment by rwmitchell** [comment]
Didn't for me:

![image](https://github.com/romkatv/powerlevel10k/assets/15944841/10f65cdc-0b1d-4603-8afe-16c93e2b8a63)

prompt shows I'm in /tmp/bar

**Comment by haolian9** [comment]
oh, sorry i failed to mention the prompt is `transient`. could you test it again with `transient prompt` on? @rwmitchell

**Comment by rwmitchell** [comment]
Same results:

![image](https://github.com/romkatv/powerlevel10k/assets/15944841/94d95d81-e822-4681-a947-f14acff9e429)


Note: my p10k was initially configured to use transient prompt:

```
3 # Wizard options: nerdfont-v3 + powerline, large icons, rainbow, unicode, 24h time,
4 # angled separators, sharp heads, flat tails, 2 lines, solid, left frame,
5 # lightest-ornaments, compact, many icons, concise, transient_prompt,
6 # instant_prompt=verbose.
```

but had disabled it to take manual control in my original screen shot, using:
```
1818 # Recommended by romkatv
1819   function p10k-on-pre-prompt()  { p10k display '1|2/right/wifi|2/left_frame'=show '2/left/time|2/right/dir'=hide }
1820   function p10k-on-post-prompt() { p10k display '1|2/right/wifi|2/left_frame'=hide '2/left/time|2/right/dir'=show }
```
For this screen shot, reset POWERLEVEL9K_TRANSIENT_PROMPT=always and commented out the two above functions.

...which doesn't show the path in the updated transient prompt.  In my initial example, entering return to move the directory name in the left prompt to the right prompt, it still shows "/private/tm/bar"

**Comment by haolian9** [comment]
sorry, i dont get you, so can you reproduce this issue or not?

**Comment by haolian9** [comment]
i just created a new user, and repeating these steps i pasted in my first comment, it does happend on my machine (tm), lol.

**Comment by rwmitchell** [comment]
No, I cannot reproduce the issue where an old directory name is displayed in any portion of the prompt.

**Comment by haolian9** [comment]
oh well, thank you anyway!

**Comment by romkatv** [comment]
Thanks for the bug report. I'll try to fix this when I get some free time. Hopefully next week.

**Comment by haolian9** [comment]
oh wow really? can you confirm it? i doubted myself and just convinced myself to let it go :joy: 
thank you for investigating it!

**Comment by romkatv** [comment]
Yes, I've reproduced it. The relevant parameter is `POWERLEVEL9K_SHORTEN_STRATEGY=truncate_to_unique`.

**Comment by romkatv** [comment]
Should be fixed now. Please update and let me know whether it works for you.

**Comment by haolian9** [comment]
after patched a69aa22, it worked! Thank you!

**Comment by romkatv** [comment]
Thanks for verifying!
